File: 181 excersises/168.cpp
#include <iostream>
using namespace std;

struct Bezout {
    int mcd, x, y;
};

Bezout euclidesExtendido(int a, int b) {
    if (b == 0) return {a, 1, 0};
    auto recur = euclidesExtendido(b, a % b);
    return {recur.mcd, recur.y, recur.x - (a/b) * recur.y};
}

int main() {
    int a, b;
    cout << "Ingrese a y b: ";
    cin >> a >> b;
    auto resultado = euclidesExtendido(a, b);
    cout << "MCD: " << resultado.mcd << endl;
    cout << "Coeficientes (x, y): (" << resultado.x << ", " << resultado.y << ")" << endl;
    return 0;
}

File: 181 excersises/168.py
def euclides_extendido(a, b):
    if b == 0: return a, 1, 0
    mcd, x1, y1 = euclides_extendido(b, a % b)
    x = y1
    y = x1 - (a // b) * y1
    return mcd, x, y

a = int(input("Ingresa a: "))
b = int(input("Ingresa b: "))
mcd, x, y = euclides_extendido(a, b)
print(f"MCD: {mcd}")
print(f"Coeficientes (x, y): ({x}, {y})")